ddtrace/tracer/tracer.go

```diff
@@ -142,22 +151,32 @@ func newTracer(opts ...StartOption) *tracer {
 		stopped:          make(chan struct{}),
 		prioritySampling: newPrioritySampler(),
 		pid:              strconv.Itoa(os.Getpid()),
+		statsd:           statsd,
 	}
-	if c.runtimeMetrics {
-		statsd, err := statsd.NewBuffered(t.config.dogstatsdAddr, 40)
-		if err != nil {
-			log.Warn("Runtime metrics disabled: %v", err)
-		} else {
-			log.Debug("Runtime metrics enabled.")
-			go t.reportMetrics(statsd, defaultMetricsReportInterval)
-		}
+	if c.runtimeMetrics && statsd != nil {
+		log.Debug("Runtime metrics enabled.")
+		go t.reportMetrics(statsd, defaultMetricsReportInterval)
 	}
-
 	go t.worker()
-
 	return t
 }
 
+func (t *tracer) statsTags() []string {
```

Comment:
I'm not really happy with this function. We use it often, and recreating tags all the time seems really inefficient. 
Maybe we can generate them once and store them for use? Then we miss any changes to `globalTags`.

Reply:
Can globalTags be modified by this point?
if not, it makes sense to have this allocated once and reused in the necessary places.

Reply:
Also might be usable if the tags are stable and known when the statsd client is created: https://godoc.org/github.com/DataDog/datadog-go/statsd#WithTags

Reply:
Right, if we're using `statsd`, we can set these up at the very beginning without doing it every time, and they will be attached automatically.